pkg/trtl/bench_test.go

```diff
@@ -0,0 +1,347 @@
+	if bench.trtl != nil {
+		if err = bench.trtl.Shutdown(); err != nil {
+			return err
+		}
+	}
+
+	// Shutdown the gRPC connection if it's running
+	if bench.grpc != nil {
+		bench.grpc.Release()
+	}
+
+	// Cleanup the benchdb and delete any stray files
+	if bench.tmpdb != "" {
+		os.RemoveAll(bench.tmpdb)
+	}
+
+	return nil
+}
+
+func BenchmarkTrtlGet(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually add some fixtures
+	// Create a put request
+	key := []byte("terminator")
+	catchphrase := []byte("hasta la vista, baby")
+	_, err := cc.Put(ctx, &pb.PutRequest{
+		Key:   key,
+		Value: catchphrase,
+	})
+	require.NoError(b, err)
+
+	// Run the trtl Get on a loop
+	b.ResetTimer()
+	var gErr error
+	for i := 0; i < b.N; i++ {
+		_, gErr = cc.Get(ctx, &pb.GetRequest{Key: key})
+	}
+	require.NoError(b, gErr)
+
+	// Run cleanup function
+	cleanup(bench)
+}
+
+func BenchmarkTrtlPut(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually create some fixtures
+	key := []byte("sarah")
+	catchphrase := []byte("How's the knee?")
+
+	// Run the trtl Get on a loop
+	b.ResetTimer()
+	var err error
+	for i := 0; i < b.N; i++ {
+		_, err = cc.Put(ctx, &pb.PutRequest{
+			Key:   key,
+			Value: catchphrase,
+		})
+	}
+	require.NoError(b, err)
+
+	// Run cleanup function
+	cleanup(bench)
+}
+
+func BenchmarkTrtlDelete(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually add some fixtures
+	var err error
+	for i := 0; i < b.N; i++ {
+		_, err = cc.Put(ctx, &pb.PutRequest{
+			Key:   []byte(fmt.Sprintf("t%d", i)),
+			Value: []byte(strconv.Itoa(i)),
+		})
+	}
+	require.NoError(b, err)
+
+	// Run the trtl Delete on a loop
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err = cc.Delete(ctx, &pb.DeleteRequest{
+			Key: []byte(fmt.Sprintf("t%d", i)),
+		})
+	}
+	require.NoError(b, err)
+
+	// Run cleanup function
+	cleanup(bench)
+}
+
+func BenchmarkTrtlIter(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually add some fixtures
+	var err error
+	for i := 0; i < b.N; i++ {
```

Comment:
Recommend:

```go
for i:= 0; i < 1000; i++
```

1000 key/value pairs feels like a good number for the iter benchmark; it's not an insignificant number of iterations  (and is above the page size) but it's not so big that it will artificially delay the request.